### ISSUE TITLE
Add an option similar to `-c` for resetting the terminal history

### ISSUE THREAD

**Issue by a-rodin** [passcod/cargo-watch#141]
The `-c` option clears the terminal, but doesn't reset the historical buffer. So running

```
cargo watch -c -s 'cargo check'
```

allows to scroll up to the previous checks, while 

```
cargo watch -s clear -s 'cargo check'
```

completely removes the history.

It is useful to have an ability to remove the history before each check if the output of `cargo check` doesn't fit into a single window. In that case scrolling up "as much us possible" always shows the beginning of the output of the most recent check, but not of past ones.

**Comment by passcod** [comment]
It wouldn't be called `-s`, as that is usually understood as `--silent`, I'd say, but I would be fine with `-C` ("strong clear") or `-cc` and `-ccc` (see below).

In more practical matters:

1. Should this call `ESC c` (full reset) or `CSI n 3` (scrollback reset)? Alternatively, should both be accessible, one with `-ccc` (full) and the other with `-cc` (scrollback)?
2. What should this do on Windows?

**Comment by a-rodin** [comment]
> Should this call `ESC c` (full reset) or `CSI n 3` (scrollback reset)? Alternatively, should both be accessible, one with `-ccc` (full) and the other with `-cc` (scrollback)?

In `tmux` where I run it only [`CSI 3 J`](https://en.wikipedia.org/wiki/ANSI_escape_code#Terminal_output_sequences) deletes the history buffer. It is interesting that `clear` sends `<esc>[H<esc>[2J<esc>[3J`:

```
$ clear | xxd
00000000: 1b5b 481b 5b32 4a1b 5b33 4a              .[H.[2J.[3J
```

> What should this do on Windows?

But on Windows it works only in an alternate buffer (`CSI ? 1049 h`). But that fact made me think that it would be actually more correct to always create an alternative buffer when `cargo watch` starts end exit from it when it exists, so that after exit from `cargo watch` it would be possible to scroll the history that happened before running `cargo watch`, similar to how for example `vim` clears the screen when starts, but after exiting it the history is available.

So I think to make it cross-platform and not damage terminal history before starting `cargo watch` I think it should be like this:

1. Print `\x1b[?1049h` on start of `cargo-watch`.
2. Print `\x1b[H\x1b[2J\x1b[3J` on each modification of files.
3. Print `\x1b[?1049l` before exit from `cargo-watch`.

Not sure about naming of the command line parameters though. With the buffers it might make sense to just change behavior of `-c` flag to the described above to not confuse the users by having multiple flags with similar functionality, although there could be reasons to have both which I don't see.

**Comment by passcod** [comment]
I like this! I like it enough that I think I want to make it default.

How about this:

- default is changed to new reset behaviour (as this is breaking, bump to v8)
- `-c, --clear` does nothing ("soft" deprecation until removal in v9)
- `--no-clear` flag disables clearing
- `--no-reset` flag or similar can be introduced later if old clearing behaviour is wanted (unlikely)

(In upstream, new behaviour is added as a new feature (non-breaking), and will be considered separately for addition to its cli flags.)

**Comment by a-rodin** [comment]
I experimented with the alternate buffer using a simple wrapper shell script

```sh
#!/bin/bash

trap "printf '\x1b[?1049l'" EXIT
printf '\x1b[?1049h'
cargo watch -s clear -s 'cargo check'
```

However, there turns out to be a huge issue with alternate buffers: the scrolling doesn't work in the alternate buffer, so if the output of the watched command is larger than one terminal screen, then its beginning becomes completely lost. And implementing scrolling in `cargo-watch` itself is possible (for example, using [`tui-rs`](https://github.com/fdehau/tui-rs)), but is a much large task than printing a few escape sequences.

So for now maybe it makes more sense to rather continue with the approach from https://github.com/passcod/cargo-watch/issues/141#issuecomment-592237246:

1. `-c` works as it works now.
2. `-cc` clears screen using `\x1b[H\x1b[2J\x1b[3J` (and on Windows it is effectively the same as `-c`).

**Comment by passcod** [comment]
Sounds good. I'll shelve the more advanced behaviour for the ever-upcoming cargo-overwatch :)

**Comment by passcod** [comment]
Closing in favour of upstream issue: https://github.com/watchexec/watchexec/issues/171.
